### DISCUSSION TITLE
some suggested small changes to manual.texinfo

### DISCUSSION THREAD

**Discussion by ko56** [shop-planner/shop3#65] [Ideas]
I've made some small tweaks, mainly some additional cross-references useful to the beginner, and fixed a typo and a typesetting problem.  I don't know how to deal with git.  How can I show you what I've done?

**Comment by rpgoldman** [comment]
If you have committed these into your git repository, you can send me a patch using `git format-patch`.  Here's a tutorial explaining how to do that:

https://www.git-tower.com/learn/git/faq/create-and-apply-patch/

That would be the best way, as in the easiest for me to apply.

Otherwise, if you haven't used git at all and haven't committed any of your changes, you could use

    git diff master -- doc/manual.texinfo > manual.patch

That would give you a patch file that you could send to me, and that I could apply to the manual.  (Note that I didn't check that command to make sure it has the right pathname to the manual.)

**Reply by ko56** [reply]
Since I haven't committed any changes with git, I used method 2, and I have a manual.patch file.

[manual.patch.gz](https://github.com/shop-planner/shop3/files/6424787/manual.patch.gz)

**Reply by ko56** [reply]
One more tweak, in sec. 4.10.
[manual.patch.gz](https://github.com/shop-planner/shop3/files/6427386/manual.patch.gz)

**Reply by rpgoldman** [reply]
I have moved your changes to Merge Request #66 

Let's discuss anything further about these changes on that Merge Request.

**Reply by ko56** [reply]
If you are OK with these changes, they are fine with me.

**Reply by rpgoldman** [reply]
Preparing this for merging led me to another round of typographical improvements (replace some misuses of `@emph` with `@var`, insert uses of `@noindent` and `@dots`, etc.).

This could go on forever, so I will merge this as long as it passes.  Please contact me off-list, so I can credit you in the Colophon for your doc improvements.

**Reply by ko56** [reply]
[I also noticed many indentation issues (needing @noindent), but I wonder if they have to do with the @lisp macro.]
Anyhow, thank you, my changes were minor.  How do I contact you off-list?

**Reply by rpgoldman** [reply]
I thought that this could be done by messaging through GitHub, but ... I can't figure out how to do that. I don't want to put my email here to be scraped so....

Would you please go to the documentation at github.io (the manual) -- or just look at the source file for the manual, and at the first page you can click on the link to my SHOP3 paper at the European Lisp Symposium. From there you will see the title page, which will give you my full name and email address. OK? It is roundabout, but it should get you what you need.

**Comment by ko56** [comment]
One more question/comment, maybe relevant to the manual.  
"Eval terms" allow general Lisp expressions, and such an expression can use arbitrary code, for example Lisp functions defined by the user, correct? (And this is a very powerful feature of SHOP3.) "call terms" also allow arbitrary functions.
Now in the definition of a domain in sec. 4.12, it says that at a minimum it will include definitions of the operators and methods. Then in sec. 4.12.1 it says that each i_i is an operator, method, or axiom.
I think it would be good to say something here about the user-defined Lisp functions, they are a rather unique feature.

**Reply by rpgoldman** [reply]
I'm not sure what we could say there, since the domain does not -- at least not in terms of syntax -- include the user-defined functions.  Those functions are parts of the lisp image that SHOP is loaded into.  The domain can *refer to* arbitrary lisp code, but that the arbitrary lisp code is part of the domain....

**Reply by ko56** [reply]
Well, you could insert somewhere, wherever you think is the right place, just what you wrote above.
(Perhaps something like this is obvious to someone familiar with Lisp, I don't know.)
Further, it seems to me that the  issue of user-defined Lisp code arises also in sec. 3.7.  Am I right? 

A separate comment: splitting sec. 4.10 is nice. The only thing left unclear is what  (forall V E L) does.

**Reply by ko56** [reply]
It seems that the (forall V E L) can be used to implement PDDL's conditional effects.

**Reply by ko56** [reply]
I am attaching another suggested set of tweaks to the manual. Revised multiple times ...
[manual.patch.gz](https://github.com/shop-planner/shop3/files/6567057/manual.patch.gz)
